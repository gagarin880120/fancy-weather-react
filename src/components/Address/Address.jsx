import React from 'react';
import PropTypes from 'prop-types';
import styles from './Address.module.css';

export default function Address({ address, countryFlagURL }) {
  console.log(countryFlagURL)
  return (
    <div className={styles.wrapper}>
      <h2 className={styles.address}>
        {address}
      </h2>
      <img src={countryFlagURL} alt="flag" className={styles.flag} />
    </div>
  );
}

Address.propTypes = {
  address: PropTypes.string,
  countryFlagURL: PropTypes.string,

};

Address.defaultProps = {
  address: '',
  countryFlagURL: '',
};
