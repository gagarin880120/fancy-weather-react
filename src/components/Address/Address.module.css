.wrapper {
  display: flex;
}

.flag {
  width: 100px;
}
