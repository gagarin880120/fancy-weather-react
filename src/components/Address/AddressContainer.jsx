import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Address from './Address';
import { selectAddress } from '../../redux/selectors';

export function AddressContainer({ address }) {
  return (
    <Address
      address={address}
    />
  );
}

export const mapStateToProps = (state) => ({
  address: selectAddress(state),
});

AddressContainer.propTypes = {
  address: PropTypes.string,

};

AddressContainer.defaultProps = {
  address: '',
};

export default connect(mapStateToProps)(AddressContainer);
