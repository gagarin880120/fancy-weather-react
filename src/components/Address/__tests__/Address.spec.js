import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Address from '../Address';

describe('Address', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<Address />);

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
});
