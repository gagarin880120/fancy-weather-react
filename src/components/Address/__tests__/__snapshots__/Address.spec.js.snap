// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address component should render without crashing 1`] = `
<div
  className="wrapper"
>
  <img
    alt="flag"
    className="flag"
    src=""
  />
  <h2
    className="address"
  >
    
  </h2>
</div>
`;
